lib/dor/text_extraction/cocina_updater.rb

```diff
@@ -62,14 +65,31 @@ def rename_document_pdf
         add_file_to_new_resource(new_filename)
       end
 
-      def add_file_to_resource(path)
-        if file_in_cocina?(path)
-          guard_overwrite(path)
+      def update_file(path)
+        resource = find_resource(path)
+
+        delete_file_from_resource(path, resource)
+        add_file_to_resource(path, resource)
+      end
+
+      def add_file(path)
+        resource = find_resource_with_stem(path)
+
+        if resource && !path.basename.to_s.match(/^#{bare_druid}\.(pdf|txt)$/)
```

Comment:
Maybe a code comment here could be useful explaining why we are ignoring #{bare_druid}.pdf and .txt  ?   

Though, actually, given the logic change above, shouldn't this just become replacements now instead of getting added over and over again?  In other words, do we even need that `!path.basename.to_s.match(/^#{bare_druid}\.(pdf|txt)$/)` check anymore?  Shouldn't our code above discover the .pdf is in cocina, and then delete and re-add it?

Reply:
Yes, updates don't care about the filename since like you said they just delete and then add from the existing resource. But when the file is being added for the first time it needs to be added to a new resource, so that's why the check is there. I will add a comment because you're right, it is super cryptic.